ta4j-core/src/main/java/org/ta4j/core/trading/rules/AndRule.java

```diff
@@ -34,32 +34,46 @@
  */
 public class AndRule extends AbstractRule {
 
-    private final Rule rule1;
-    private final Rule rule2;
+    private final Rule[] rules;
 
     /**
      * Constructor
      *
-     * @param rule1 a trading rule
-     * @param rule2 another trading rule
+     * @param rules all trading rules
      */
-    public AndRule(Rule rule1, Rule rule2) {
-        this.rule1 = rule1;
-        this.rule2 = rule2;
+    public AndRule(Rule... rules) {
+        if (rules == null || rules.length < 2) {
```

Comment:
should be 

`rules == null || rules[0] == null || rules.length < 2`

because the array can contain a null value, which must be considered.